web/src/pages/Leaderboard.js

```diff
@@ -9,12 +9,19 @@ export default function Leaderboard() {
   const navigate = useNavigate();
 
   const convertData = (obj, type, deltaObj) => {
-    return Object.entries(obj).map(([key, value], index) => ({
-      id: (index + 1).toString(),
-      name: key.replace(`${type}_`, ""),
-      score: value.toFixed(4),
-      improved: deltaObj[key] > 0.0 ? 1 : deltaObj[key] < 0.0 ? -1 : 0,
-    }));
+    const emailPattern = /^(.*?)(@gmail\.com|@yahoo\.com|@hotmail\.com|@outlook\.com|@icloud\.com)?$/;
```

Comment:
hmm this only works for a few email providers no? e.g. would xxx@berkeley.edu work?

Reply:
Would it be best to look for the @ symbol, then some domain, and then some ending like ".edu" or ".com", etc.?

Reply:
this can be helpful
https://chatgpt.com/share/671fef39-9640-8003-83cc-b69e9b3de75a

Reply:
thanks! I updated the pattern using this